Небольшие исправления настроек проекта

art4vm opened the pull request:
Изменен параметр нормализации окончаний строк при checkin в git
 Теперь бинарные файлы игнорируются
 Возвращены рабочие спрайты для персонажей

Исправлен регистр каталогов Button и Copyright
Исправлен каталог сборки приложения
Добавлен алиас для абсолютных путей в сборку webpack

jarith commented on app/src/components/Button/Button.tsx:
```diff
@@ -0,0 +1,15 @@
+import React, {ButtonHTMLAttributes, memo} from 'react';
+import './Button.scss';
+
+function component(props: ButtonHTMLAttributes<HTMLButtonElement>): JSX.Element {
```
Кажется, писал в одном из прошлых PR, что можно

```suggestion
function component({type = 'button', className = 'btn t-main}: ButtonHTMLAttributes<HTMLButtonElement>): JSX.Element {
```
и `defaultProps` не нужны

Вместо `ButtonHTMLAttributes` стоит объявить свой тип для пропсов, где явно описать все передаваемые атрибуты

art4vm replied:
Я не добавлял этот файл, а переименовал каталог. Исправлю чтобы это было переименование

> defaultProps не нужны

В данном случае это немного не удобно, тк используется ...rest, который мы передаем в дефолтный button, и теперь придется явно еще 2 параметра передавать.

> Вместо ButtonHTMLAttributes стоит объявить свой тип для пропсов, где явно описать все передаваемые атрибуты

У этого компонента нет своих аттрибутов, это обертка над стандартной кнопкой с дефолтными классами

art4vm committed:
Небольшие исправления настроек проекта

Изменен параметр нормализации окончаний строк при checkin в git
Теперь бинарные файлы игнорируются
Возвращены рабочие спрайты для персонажей
Исправлен регистр каталогов Button и Copyright
Исправлен каталог сборки приложения
Добавлен алиас для абсолютных путей в сборку webpack
